java/src/main/java/com/anaconda/skein/MsgUtils.java

```diff
@@ -312,10 +318,16 @@ public static LocalResource readFile(Msg.File f) {
     for (Map.Entry<String, Msg.Service> entry : spec.getServicesMap().entrySet()) {
       services.put(entry.getKey(), readService(entry.getValue()));
     }
+    final List<Path> nameNodes = Lists.transform(
+        spec.getNameNodesList(),
+        new Function<String, Path>() {
+          public Path apply(String uri) { return new Path(uri); }
+        });
```

Comment:
Any reason not to do (untested):

```java
List<Path> nameNodes = new ArrayList<Path>(spec.getNameNodesCount());
for (String uri : spec.getNameNodesList()) {
  nameNodes.add(new Path(uri));
}
```

I find this version more readable and concise, but admit it's just a matter of style. I am quite new at Java, and find the deep nesting needed here to do what amounts to a `name_nodes = map(Path, name_node_uris)` harder to read than manually writing out the loop. Perhaps my opinion on this will change over time.

Not a strong opinion though, feel free to keep it as is.

Reply:
The deep nesting comes from lack of lambdas in Java prior to version 8. In Java 8 this would look like

```java
Lists.transform(..., uri -> new Path(uri))
```

which is clearly much more readable :) which brings us back to Java 8 or Kotlin (#41) for improving the development experience.